index.js

```diff
@@ -105,16 +113,28 @@ class PearIPC extends ReadyResource {
     this.stream.on('close', onclose)
 
     this._register()
+
+    if (this.server === null && this.id === -1) {
+      this._heartbeat = setInterval(() => {
+        this._beat()
+      }, HEARTBEAT_INTERVAL)
+      await this._beat()
+    }
+  }
+
+  async _beat () {
+    const result = await this._ping()
+    if (result.beat !== 'pong') this.close()
```

Comment:
Sanity check: what's the timeout on an IPC call, or what happens if it fails? The code seems to assume _ping can never throw , and will always resolve in a reasonable time (if it can take longer than HEARTBEAT_INTERVAL, it's a bit strange)

Reply:
Actually, the `this._heartbeat` interval already takes care of closing the clients that do not respond to the heartbeat. So I think calling close here isn't needed--or are those 2 different flows?

Reply:
@HDegroote good point, it can fail on connection failure, which is after 20s - to avoid unhandleds I've modified so the ping call is try/catch'ed